src/auth/permissions/index.ts

```diff
@@ -0,0 +1,173 @@
+import * as knex from 'knex';
+import { PermissionsInitialiser, GetPermissionIds, GetRoleId } from './permissions';
+
+const getPermissionIds: GetPermissionIds = ({ client, resource, permissionLevel, access }) =>
+  client
+    .select('permission_id')
+    .table('permission')
+    .where({
+      ['permission_entity']: resource,
+      ['permission_level']: permissionLevel,
+      ['access_type']: access,
+    });
+
+const getRoleId: GetRoleId = async({ client, role }) =>
+  client
+    .select('access_role_id')
+    .table('access_role')
+    .where({ ['access_role_name']: role })
+    .then((res) => res[0].access_role_id);
+
+const permissionsInitialiser: PermissionsInitialiser = (config) => {
+  const client = knex(config);
```

Comment:
I think it's better to take the actual `knex` client as an argument here, because we'll want to share the same connection pool across the entire app.

It's different from the e-mail service, because the postmark client (AFAIK) doesn't create any remote connections when it's instantiated. Everything is done via HTTP requests to the postmark server, and only on method calls.